src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -359,6 +362,7 @@ public override void OnDismiss(IDialogInterface dialog)
 			{
 				_modal.PropertyChanged -= OnModalPagePropertyChanged;
 				_modal.HandlerChanged -= OnPageHandlerChanged;
+				Dialog?.SetOnKeyListener(null);
```

Comment:
Alright, we're getting close

Instead of using the keylistener can you just override all the available OnKey* Methods